bin/node/src/service.rs

```diff
@@ -212,14 +212,28 @@ fn setup(
     (
         RpcHandlers,
         Arc<NetworkService<Block, <Block as BlockT>::Hash>>,
+        ProtocolNaming,
         NetworkStarter,
     ),
     ServiceError,
 > {
+    let genesis_hash = client
+        .block_hash(0)
+        .ok()
+        .flatten()
+        .expect("we should have a hash");
+    let chain_prefix = match config.chain_spec.fork_id() {
+        Some(fork_id) => format!("/{}/{}", genesis_hash, fork_id),
```

Comment:
what's the idea with `fork_id`?

Reply:
This is for when/if we ever hard fork. We don't expect this to happen and afaik `fork_id` currently returns `None`, so all good, but if it ever happens we definitely don't want the networks mixing.

Reply:
then I'd say we should either not consider this case in this PR at all, or include the non-fork name in the fallback list

Reply:
The latter completely misses the point, anything included in the fallback list _will_ be connected, and we don't want that.

As for the former, we could, but I am trying to stay completely in sync with the substrate naming convention, as I think in this case it's smart. The main reason is that if there ever is a fork it might be hard to remember that this is one of the things that need to be changed for stuff to work (or worse, someone hacks around the problem).

Reply:
so why do you want to have fallback names now, but not in case of a hard fork? both situations require a transition period during which everyone updates their nodes, right?

Reply:
Oh, nono, a hard fork would start completely disconnected from the previous network. This is _extremely_ unlikely to ever happen, but in case it does there will be no transition period like this.